truffleruby/src/main/ruby/core/truffle/cext.rb

```diff
@@ -1111,6 +1111,53 @@ def rb_proc_new(function, value)
     end
   end
 
+  def foreign_call_with_block(function, *args, &block)
+    Truffle::Interop.execute(function, *args)
+  end
+
+  def rb_protect_with_block(function, arg, block)
+    res = nil;
+    pos = 0;
+    e = store_exception do
+      puts "Block is #{block}"
+      res = foreign_call_with_block(function, arg, &block)
+    end
+    unless e == nil
+      store = Thread.current[:__stored_exceptions__]
+      store = Thread.current[:__stored_exceptions__] = [] unless store
+      pos = store.push(e).size
+    end
+    [res, pos]
+  end
+
+  def rb_protect(function, arg)
```

Comment:
Can this reuse rb_protect_with_block above with `block=nil`? Or just add a `if block` in the store_exception block?